packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -32,24 +33,52 @@ enum Orientation {
 /// exception, unless the `nullOk` argument is set to true, in which case it
 /// returns null.
 ///
-/// MediaQueryData includes two [EdgeInsets] values:
-/// [padding] and [viewInsets]. These
-/// values reflect the configuration of the device and are used by
-/// many top level widgets, like [SafeArea] and the Cupertino and
-/// Material scaffold widgets. The padding value defines areas that
-/// might not be completely visible, like the display "notch" on the
-/// iPhone X. The viewInsets value defines areas that aren't visible at
-/// all, typically because they're obscured by the device's keyboard.
+/// ## Insets and Padding
 ///
-/// The viewInsets and padding values are independent, they're both
-/// measured from the edges of the MediaQuery widget's bounds. The
-/// bounds of the top level MediaQuery created by [WidgetsApp] are the
-/// same as the window that contains the app.
+/// ![A diagram of padding, viewInsets, and viewPadding in correlation with each
+/// other](https://flutter.github.io/assets-for-api-docs/assets/widgets/media_query.mp4)
 ///
-/// Widgets whose layouts consume space defined by [viewInsets] or
-/// [padding] should enclose their children in secondary MediaQuery
+/// This diagram illustrates how [padding] relates to [viewPadding] and
+/// [viewInsets], shown here in its simplest configuration, as the difference
+/// between the two. In cases when the viewInsets exceed the viewPadding, like
+/// when a keyboard is shown below, padding goes to zero rather than a negative
+/// value. Therefore, padding is calculated by taking
+/// `max(0.0, viewPadding - viewInsets)`.
+///
+/// {@animation 300 300 https://flutter.github.io/assets-for-api-docs/assets/widgets/window_padding.mp4}
+///
+/// In this diagram, the black areas represent system UI that the app cannot
+/// draw over. The red area represents view padding that the application may not
+/// be able to detect gestures in and may not want to draw in. The grey area
+/// represents the system keyboard, which can cover over the bottom view padding
+/// when visible.
+///
+/// MediaQueryData includes three [EdgeInsets] values:
+/// [padding], [viewPadding], and [viewInsets]. These values reflect the
+/// configuration of the device and are used by many top level widgets, listed
```

Comment:
`man topy level widgets` seems a bit vague.  Maybe it would be better to say why a widget would use these, e.g. `are used and optionally consumed by widgets that position content within these insets.`  Perhaps with a sentence later in this paragraph explaining that a widget may consume the available insets by creating a new `MediaQuery` with methods like `removePadding` etc.

Reply:
(scratch adding it later in the paragarph - I see it a couple paragraphs down from here and that seems fine)